File: AssayDashboard/Source/Hologic.AssayDashboard.Client/src/app/assayviewrepo/assayviewrepo-models/assay.ts
/*
// ----------------------------------------------------------------
// Copyright (c) 2017 Hologic, Inc.  All Rights Reserved.
// ----------------------------------------------------------------
*/

import { DataReduction } from "app/assayviewrepo/assayviewrepo-models/datareduction";
import { Reagent } from "app/assayviewrepo/assayviewrepo-models/reagent";
import { Control } from "app/assayviewrepo/assayviewrepo-models/control";
import { Calibrator } from "app/assayviewrepo/assayviewrepo-models/calibrator";
import { Analyte } from "app/assayviewrepo/assayviewrepo-models/analyte";
import { ReagentTypeDescription } from "app/assayviewrepo/assayviewrepo-models/reagentTypeDescription";
import { AssayMessage } from "app/assayviewrepo/assayviewrepo-models/assaymessage";

export class Assay {
  assayName: string;
  version: string;
  assayStatus: string;
  assayTypeID: string;
  thermalProfileReference: string;
  frontEndReference: string;
  resource: string;
  objectCrc: number;
  isFrontEnd: string;
  isLocked: string;
  calculatedCrc: number;
  analytes: string;
  analyteList: Analyte[];
  reagentTypeDescription: ReagentTypeDescription;
  fileName: string;
  type: string;
  dataReduction: DataReduction;
  xmlText: string = ''; 
  processSteps: any[];
  assayMessages: AssayMessage[];
  date: number;

  equals(assay: Assay): boolean {
    if (assay.fileName === this.fileName) 
      return true;
    return false;
  }

  getType(): void {
    if (this.isLocked === 'true' && this.frontEndReference === undefined && this.isFrontEnd === undefined)
      this.type = "TMA";
    if (this.isLocked === 'true' && this.frontEndReference !== undefined && String(this.assayTypeID).length <= 3)
      this.type = "PCR";
    if (this.isLocked === 'true' && this.isFrontEnd === 'true')
      this.type = "Front-end";
    if (String(this.assayTypeID).length > 3 && this.isLocked === 'false')
      this.type = "Open Access";
  }
}

File: AssayDashboard/Source/Hologic.AssayDashboard.AssayParser/Models/Assay.cs
using System;
using System.Collections.Generic;
using System.Xml.Serialization;

namespace Hologic.AssayDashboard.AssayParser
{
    [XmlRoot("AssayDescription")]
    public class Assay
    {
        [XmlElement("AssayShortName")]
        public string assayName;

        [XmlElement("Version")]
        public string version;

        [XmlElement("AssayTypeID")]
        public string assayTypeID;

        [XmlElement("AssayStatus")]
        public string assayStatus;

        [XmlElement("ThermalProfileReference")]
        public string thermalProfileReference;

        [XmlElement("FrontEndReference")]
        public string frontEndReference;

        [XmlElement("IsFrontEnd")]
        public string isFrontEnd;

        [XmlElement("IsLocked")]
        public string isLocked;

        [XmlArray("AnalyteList")]
        public List<Analyte> analyteList;

        [XmlArray("AssayMessages")]
        public List<AssayMessage> assayMessages;

        [XmlElement("ReagentTypeDescriptions")]
        public ReagentTypeDescription reagentTypeDescription;

        [XmlIgnore]
        public List<Dictionary<string, Object>> processSteps;

        public ushort calculatedCrc;
        public string type;
        public string analytes;
        public string xmlText;

        [XmlElement("DataReductionMethod")]
        public DataReduction dataReduction;

        [XmlAttribute("ObjectCRC")]
        public ushort objectCrc;

        [XmlAttribute("Resource")]
        public string resource;

        public Assay()
        {
            this.assayName = "";
            this.version = "";
            this.assayTypeID = "";
            this.assayStatus = "";
            this.thermalProfileReference = "";
            this.frontEndReference = "";
            this.isFrontEnd = "";
            this.isLocked = "";
            this.type = "";
            this.analytes = "";
            this.xmlText = "";
            this.analyteList = new List<Analyte>();
            this.assayMessages = new List<AssayMessage>();
            this.reagentTypeDescription = null;
            this.dataReduction = null;
        }

        public void getType() {
            if (this.isLocked == "true" && this.frontEndReference.Length == 0 && this.isFrontEnd.Length == 0)
            {
              this.type = "TMA";
            }
            if (this.isLocked == "true" && this.frontEndReference.Length != 0 && this.assayTypeID.Length <= 3)
            {
              this.type = "PCR";
            }
            if (this.isLocked == "true" && this.isFrontEnd == "true")
            {
              this.type = "Front-end";
            }
            if (this.assayTypeID.Length > 3 && this.isLocked == "false")
            {
              this.type = "Open Access";
            }
        }
    }
}
